modules/placement-driver/src/main/java/org/apache/ignite/internal/placementdriver/leases/LeaseTracker.java

```diff
@@ -181,6 +168,8 @@ public CompletableFuture<Void> onUpdate(WatchEvent event) {
                             primaryReplicaWaiters
                                     .computeIfAbsent(grpId, groupId -> new PendingIndependentComparableValuesTracker<>(MIN_VALUE))
                                     .update(lease.getExpirationTime(), lease);
+
+                            fireEventFutures.add(fireEventReplicaBecomePrimary(event.revision(), lease));
```

Comment:
Hmm, I believe that we will fireEventReplicaBecomePrimary on lease prolongation which seems incorrect. By the way, let's add test for that.